searches/fibonacci_search.py

```diff
@@ -1,51 +1,100 @@
-# run using python fibonacci_search.py -v
-
 """
-@params
-arr: input array
-val: the value to be searched
-output: the index of element in the array or -1 if not found
-return 0 if input array is empty
+This is pure Python implementation of fibonacci search.
+
+Resources used:
+https://en.wikipedia.org/wiki/Fibonacci_search_technique
+
+For doctests run following command:
+python3 -m doctest -v fibonacci_search.py
+
+For manual testing run:
+python3 fibonacci_search.py
 """
+import sys
+from functools import lru_cache
+
+
+@lru_cache
+def fibonacci(k: int) -> int:
+    """Finds fibonacci number in index k.
 
+    Parameters
+    ----------
+    k : int
+        Index of fibonacci.
```

Comment:
```suggestion
    Parameters
    ----------
    k : Index of fibonacci.
```
Already mentioned in type hints.